### PR TITLE
Fix parse errors handling in BDD

### PR DESCRIPTION
**Pull request by dmikhalin**
## What is the goal of this PR?

We improved parse errors handling in the `behaviour` test package.

## What are the changes implemented in this PR?

We added asserts at the all stages of query processing for "`typeql define [...]`" and "`typeql insert [...]`" expressions.

### CODE REVIEW

File: tests/behaviour/typeql/steps.rs
Comment (dmitrii-ubskii) [inline tests/behaviour/typeql/steps.rs:86]: Consider matching inline:
```suggestion
            match context.transaction().query().insert(&parsed.unwrap().to_string()) {
```

File: tests/behaviour/typeql/steps.rs
Comment (dmitrii-ubskii) [inline tests/behaviour/typeql/steps.rs:87]: `unwrapped` is not a meaningful name. The `Ok()` variant contains a stream, the name should reflect that.

**Review by dmitrii-ubskii — CHANGES_REQUESTED** [review]

File: tests/behaviour/typeql/steps.rs
Comment (dmitrii-ubskii) [inline tests/behaviour/typeql/steps.rs:94]: It's a bit of a smell that we perform the same check here repeatedly. This can be streamlined. Look into `Result` combinators such as `and_then()`.

**Review by dmitrii-ubskii — CHANGES_REQUESTED** [review]

File: tests/behaviour/typeql/steps.rs
Comment (dmitrii-ubskii) [inline tests/behaviour/typeql/steps.rs:52]: This async / `.map_err` / `.and_then` chaining warrants a comment for future reference, even though we've talked through it on a call.

Comment (dmikhalin) [reply tests/behaviour/typeql/steps.rs:52]: We replaced `match`es with the combinators chaining to avoid repetitive exception checks.
We convert all errors (of incompatible types) to `String`s because they must have the same type.
The result of query parsing must be a `Future` because we apply the `async` function (using `and_then`) that returns a `Future` to it.

File: tests/BUILD
Comment (flyingsilverfin) [inline tests/BUILD:41]: A good ordering of dependencies would by "this" repository first, then "vaticle" dependencies, then "crate" dependencies!

Comment (flyingsilverfin) [reply tests/BUILD:41]: cc @dmitrii-ubskii  for you to take not of for yourself too ;) This is the level of detail I'm going to be pushing you toward over time

File: tests/behaviour/typeql/steps.rs
Comment (flyingsilverfin) [inline tests/behaviour/typeql/steps.rs:61]: `stream` is not a very good name - it doesn't give you any context for what it represents (only the type, which the IDE should already tell you!). 
I'd say this should be 'answers' or 'inserted'. 
cc @dmitrii-ubskii  since you wrote the original version on the left

Comment (flyingsilverfin) [reply tests/behaviour/typeql/steps.rs:61]: Applies throughout

**Review by flyingsilverfin — APPROVED** [review]
I've approved pending two minor changes you can make before merging :) I don't need to review again, I'll trust you to follow through on the comments made.

File: tests/behaviour/typeql/steps.rs
Comment (flyingsilverfin) [reply tests/behaviour/typeql/steps.rs:61]: note: `answers` is only the right name if it actually returns answers! If it's just a `Result` of nothing, then `result` is an appropriate name

File: tests/BUILD
Comment (dmitrii-ubskii) [reply tests/BUILD:41]: That's the opposite of how we've been doing it so far jsyk.

Comment (flyingsilverfin) [reply tests/BUILD:41]: You can fix it at some point :) no rush atm

File: tests/behaviour/typeql/steps.rs
Comment (flyingsilverfin) [reply tests/behaviour/typeql/steps.rs:61]: Right - I was saying that `result` is worse than `stream` which is worse than `answers` or `inserted`. However, for `define` queries for example that have a void return type and are just a checked result, `result` is the best name I think.

Comment (dmikhalin) [reply tests/behaviour/typeql/steps.rs]: Renamed to `inserted`.

File: tests/BUILD
Comment (dmikhalin) [reply tests/BUILD]: Reordered

**Review by dmitrii-ubskii — APPROVED** [review]

**Pull request merged**
